spec/controllers/exercises_controller_spec.rb

```diff
@@ -1,92 +1,212 @@
 require 'rails_helper'
 
 RSpec.describe ExercisesController, type: :controller do
-  describe '#show' do
-    let(:user) { FactoryBot.build_stubbed(:user) }
+  describe '#index' do
+    let(:exercise) { FactoryBot.create(:exercise, user_id: current_user.id) }
+    let(:current_user) { FactoryBot.create(:user) }
 
-    before do
-      allow(controller).to receive(:authenticate_user!)
-      allow(controller).to receive(:current_user).and_return(user)
-    end
+    context 'As an authenticated user' do
+      it 'returns the index successfully (200)' do
+        allow(controller).to receive(:authenticate_user!)
+        allow(controller).to receive(:current_user).and_return(current_user)
 
-    it 'assigns the exercises ivar to the exercise for an id' do
-      # Arrange, Act, Assert
-      exercise = FactoryBot.create(:exercise)
-      get :show, id: exercise.id
-      exercise_ivar = assigns(:exercise)
-      expect(exercise_ivar).to eq(exercise)
+        get :index
+
+        aggregate_failures do
+          expect(response).to be_successful
+          expect(response).to have_http_status '200'
+        end
+      end
     end
 
-    it 'raises an active record error if the exercise does not exist' do
-      expect { get :show, id: 'invalid ID' }.to raise_error(ActiveRecord::RecordNotFound)
+    context 'As a guest' do
+      it 'redirects you to the sign-in / sign-up page' do
+        get :index
+        expect(response).to redirect_to '/users/sign_in'
+      end
+
+      it 'returns a 302 (redirect) response' do
+        get :index
+        expect(response).to have_http_status '302'
+      end
     end
+  end
+
+  describe '#show' do
+    let(:exercise) { FactoryBot.create(:exercise, user_id: current_user.id) }
+    let(:current_user) { FactoryBot.create(:user) }
+
+    context 'when logged in' do
+      before do
+        allow(controller).to receive(:authenticate_user!)
+        allow(controller).to receive(:current_user).and_return(current_user)
+      end
+
+      it 'responds successfully to a GET request' do
+        get :show, params: { id: exercise.id }
+        expect(response).to be_successful
+      end
 
-    it 'assigns the return value of exercise#rehearsals_for_user to the rehearsals ivar' do
-      exercise = FactoryBot.create(:exercise)
-      allow(Exercise).to receive(:find).with(exercise.id.to_s).and_return(exercise)
+      it 'returns a 200 HTTP response code' do
+        get :show, params: { id: exercise.id }
+        expect(response).to have_http_status('200')
+      end
```

Comment:
Great - thanks for showing that snippet. That is a great point. 

The reason I used this two-test pattern is because that's how Aaron Sumner organises all of his Controller Specs: https://github.com/everydayrails/everydayrails-rspec-2017/blob/master/spec/controllers/projects_controller_spec.rb#L10

I don't know. I'm fine with trusting your reasoning here. Your point and evidence here on `successful?` is strong.

So therefore: I deleted the second test in all of these contexts.